xchange-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioStreamingService.java

```diff
@@ -0,0 +1,134 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+
+public class CexioStreamingService extends JsonNettyStreamingService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CexioStreamingService.class);
+
+    public static final String CONNECTED = "connected";
+    public static final String AUTH = "auth";
+    public static final String PING = "ping";
+    public static final String PONG = "pong";
+    public static final String ORDER = "order";
+
+    private String apiKey;
+    private String apiSecret;
+
+    public CexioStreamingService(String apiUrl) {
+        super(apiUrl, Integer.MAX_VALUE);
+    }
+
+    @Override
+    protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+        return null;
+    }
+
+    @Override
+    public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+        return null;
+    }
+
+    @Override
+    public String getUnsubscribeMessage(String channelName) throws IOException {
+        return null;
+    }
+
+    @Override
+    public void messageHandler(String message) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode;
+        try {
+            jsonNode = objectMapper.readTree(message);
+        } catch (IOException e) {
+            LOG.error("Error parsing incoming message to JSON: {}", message);
+            return;
+        }
+        handleMessage(jsonNode);
+    }
+
+    @Override
+    protected void handleMessage(JsonNode message) {
+        LOG.debug("Receiving message: {}", message);
+        JsonNode cexioMessage = message.get("e");
+
+        if (cexioMessage != null) {
+            switch (cexioMessage.textValue()) {
+                case CONNECTED:
+                    auth();
+                    break;
+                case AUTH:
+                    CexioWebSocketAuthResponse response =
+                            (CexioWebSocketAuthResponse)deserialize(message, CexioWebSocketAuthResponse.class);
+                    if ((response != null) && !response.isSuccess()) {
+                        LOG.error("Authentication error: {}", response.getData().getError());
+                    }
+                    break;
+                case PING:
+                    pong();
+                    break;
+                case ORDER:
+                    CexioWebSocketOrderMessage order =
+                            (CexioWebSocketOrderMessage)deserialize(message, CexioWebSocketOrderMessage.class);
+                    if (order != null) {
+                        if (order.getData().isCancel()) {
+                            LOG.info(String.format("Order is cancelled: %s", order.getData()));
+                        } else if (order.getData().getRemains().compareTo(BigDecimal.ZERO) == 0) {
+                            LOG.info(String.format("Order is fully filled: %s", order.getData()));
```

Comment:
Order can be partially filled. It's a regular situation, and have to be logged too.